cmd/spire-agent/cli/run/run.go

```diff
@@ -27,30 +28,32 @@ const (
 	defaultSocketPath = "./spire_api"
 
 	// TODO: Make my defaults sane
-	defaultDataDir   = "."
-	defaultLogLevel  = "INFO"
-	defaultPluginDir = "conf/agent/plugin"
-
-	defaultUmask = 0077
+	defaultDataDir  = "."
+	defaultLogLevel = "INFO"
+	defaultUmask    = 0077
 )
 
-// Struct representing available configurables for file and CLI
-// options
-type RunConfig struct {
-	ServerAddress   string
-	ServerPort      int
-	TrustDomain     string
-	TrustBundlePath string
-	JoinToken       string
-
-	SocketPath string
-	DataDir    string
-	PluginDir  string
-	LogFile    string
-	LogLevel   string
+// RunConfig represents the available configurables for file
+// and CLI options
+type runConfig struct {
+	AgentConfig    agentConfig                                   `hcl:"agent"`
+	PluginsConfigs map[string]map[string]catalog.HclPluginConfig `hcl:"plugins"`
```

Comment:
perhaps this warrants a new type? `PluginConfigMap`?

Reply:
Certainly, thanks!